.ci/watchdog/Watchdog.py

```diff
@@ -0,0 +1,359 @@
+        # Jenkins Wrapper object for CI job
+        self._jenkins = JenkinsWrapper(jenkins_token, jenkins_user=jenkins_user, jenkins_server=jenkins_server)
+        # Load GitHub token and log in, retrieve pull requests
+        self._git = GitWrapper(git_token, repository='NervanaSystems', project='ngraph-onnx')
+        # Create Slack api object
+        self._slack_app = SlackCommunicator(slack_token)
+        self._ci_job_name = ci_job_name
+        self._watchdog_job_name = watchdog_job_name
+        # Read config file
+        self._config = self._read_config_file()
+        # Time at Watchdog initiation
+        self._now_time = self._git.get_git_time()
+
+    def run(self):
+        """Runs watchdog main logic. 
+        
+        Retrievs list of pull requests and passes it to the method responsible for checking them.
+        """
+        try:
+            pull_requests = self._git.get_pull_requests()
+        except:
+            message = "Failed to retrieve Pull Requests!"
+            log.exception(message)
+            self._queue_message(message, message_severity=999)
+        try:
+            self._check_prs(pull_requests)
+        except Exception as e:
+            log.exception(str(e))
+            self._queue_message(str(e), message_severity=999)
+        self._send_message()
+
+    def _read_config_file(self):
+        """Reads Watchdog config file stored on the system. 
+        
+        The file stores every fail already reported along with 
+        timestamp. This mechanism is used to prevent Watchdog from reporting same failure multiple times. In case 
+        there's no config under the expected path, appropriate data structure is created and returned.
+
+            :return:            Returns dict of dicts with reported fails with their timestamps
+            :rtype:             dict of dicts
+        """
+        
+        if os.path.isfile(_CONFIG_PATH):
+            log.info("Config file exists, reading.")
+            file = open(_CONFIG_PATH,'r')
+            data = json.load(file)
+        else:
+            log.info("No config file.")
+            data = { _PR_REPORTS_CONFIG_KEY: {} }
+        return data
+
+    def _check_prs(self, pull_requests):
+        """ 
+        Loops through pull requests, retrieving list of statuses for every PR's last commit. Filters out statuses 
+        unrelated to nGraph-ONNX Jenkins CI and passes relevant statuses to method that interprets them. If no 
+        commit statuses related to Jenkins are availible after time defined by**_AWAITING_JENKINS_TRESHOLD** reports
+        fail.
+        This method also updates Watchdog config with current Pull Requests.
+
+            :param pull_requests:       Paginated list of Pull Requests
+            :type pull_requests:        github.PaginatedList.PaginatedList of github.PullRequest.PullRequest
+        """
+        
+        current_prs = []
```

Comment:
please add separate PR for refactoring this method, we can discuss with @postrational how it might looks like